libyul/backends/evm/EVMDialect.h

```diff
@@ -62,25 +62,39 @@ struct BuiltinFunctionForEVM: public BuiltinFunction
  * The main difference is that the builtin functions take an AbstractAssembly for the
  * code generation.
  */
-struct EVMDialect: public Dialect
+struct EVMDialect: Dialect
 {
 	/// Constructor, should only be used internally. Use the factory functions below.
 	EVMDialect(langutil::EVMVersion _evmVersion, std::optional<uint8_t> _eofVersion, bool _objectAccess);
 
-	/// @returns the builtin function of the given name or a nullptr if it is not a builtin function.
-	BuiltinFunctionForEVM const* builtin(YulName _name) const override;
+	/// @returns the builtin function of the given name or null if it is not a builtin function.
+	std::optional<BuiltinHandle> findBuiltin(std::string_view _name) const override;
+
+	BuiltinFunctionForEVM const& builtin(BuiltinHandle const& handle) const override
+	{
+		if (isVerbatimHandle(handle))
+		{
+			auto const& verbatimFunctionPtr = m_verbatimFunctions.at(handle.id);
+			yulAssert(verbatimFunctionPtr);
+			return *verbatimFunctionPtr;
+		}
+
+		auto const& maybeBuiltin = m_functions.at(handle.id - verbatimIdOffset);
```

Comment:
Also, this function is a bit too long to be in the header. It should be in the .cpp file, unless you measured it and know that the performance hit from it not being inlined actually matters.

I actually had trouble finding it at first because I didn't expect to find something like this here :)

Reply:
> Also, this function is a bit too long to be in the header. It should be in the .cpp file, unless you measured it and know that the performance hit from it not being inlined actually matters.
> 
> I actually had trouble finding it at first because I didn't expect to find something like this here :)

Yeah that is fair. In my mind it was a simple map / vector lookup. But with all the error handling and verbatim business, it's not so simple anymore. I'll double check with the benchmarks that it's no significant performance hit - although I doubt that it would be. Fixed.